src/temgymbasic/components.py

```diff
@@ -240,15 +389,33 @@ class ParallelBeam(Source):
     def __init__(
         self,
         z: float,
-        radius: float = None,
+        radius: float,
+        wavelength: Optional[float] = 0.0,
+        phi_0: Optional[float] = 0.0,
         tilt_yx: Tuple[float, float] = (0., 0.),
         name: Optional[str] = None,
     ):
         super().__init__(z=z, tilt_yx=tilt_yx, name=name)
         self.radius = radius
 
+        # Check if both wavelength and acceleration_voltage are provided
```

Comment:
As above, also code duplication ?

Reply:
Yes... I will think about how to deal with this.